pkg/api-server/resource_endpoints.go

```diff
@@ -981,13 +984,67 @@ func (r *resourceEndpoints) rulesForResource() restful.RouteFunction {
 		sort.Slice(httpMatches, func(i, j int) bool {
 			return httpMatches[i].Hash < httpMatches[j].Hash
 		})
+		outbounds, err := r.outboundsForDPP(request.Request.Context(), dp, meshName)
+		if err != nil {
+			rest_errors.HandleError(request.Request.Context(), response, err, "Failed to get outbounds")
+			return
+		}
 		out := api_types.InspectRulesResponse{
 			HttpMatches: httpMatches,
 			Resource:    oapi_helpers.ResourceToMeta(resource),
 			Rules:       rules,
+			Outbounds:   outbounds,
 		}
 		if err := response.WriteAsJson(out); err != nil {
 			rest_errors.HandleError(request.Request.Context(), response, err, "Failed writing response")
 		}
 	}
 }
+
+func (r *resourceEndpoints) outboundsForDPP(
+	ctx context.Context,
+	dpp *core_mesh.DataplaneResource,
+	meshName string,
+) ([]api_common.Outbound, error) {
+	meshContext, err := r.meshContextBuilder.BuildIfChanged(ctx, meshName, nil)
```

Comment:
Do we get the full MeshContext again right? That's problematic then because the whole point of the `_rules` endpoint is that it didn't require getting all DPPs because this didn't scale at all in global as each of this calls would be in `O(#Dpp)`.

If we need to break this then we'd need to rework the MeshContextBuillder to make this more workable, in which case we should punt this to 2.10.x